.github/workflows/rust.yml

```diff
@@ -65,42 +65,73 @@ jobs:
           # this key equals the ones on `linux-build-lib` for re-use
           key: cargo-cache-benchmark-${{ hashFiles('datafusion/**/Cargo.toml', 'benchmarks/Cargo.toml', 'datafusion-cli/Cargo.toml') }}
 
-      - name: Check workspace without default features
+      - name: Check datafusion without default features
+        # Some of the test binaries require the parquet feature still
+        #run: cargo check --all-targets --no-default-features -p datafusion
         run: cargo check --no-default-features -p datafusion
 
       - name: Check datafusion-common without default features
-        run: cargo check --tests --no-default-features -p datafusion-common
+        run: cargo check --all-targets --no-default-features -p datafusion-common
+
+      - name: Check datafusion-functions
+        run: cargo check --all-targets --no-default-features -p datafusion-functions
 
       - name: Check workspace in debug mode
         run: cargo check
 
-      - name: Check workspace with all features
+      - name: Check workspace with avro,json features
         run: cargo check --workspace --benches --features avro,json
 
+      - name: Check Cargo.lock for datafusion-cli
+        run: |
+          # If this test fails, try running `cargo update` in the `datafusion-cli` directory
+          # and check in the updated Cargo.lock file.
```

Comment:
`cargo update` might be "overkill" for this kind of error. It will try to update all the dependence to the latest available version that meets the semantic version rule, and generate some not intended diff. Usually a `cargo check` that toggles `cargo` to read `Cargo.toml` file can update the lock file.

But I'm OK with `cargo update` either, as the dependency should be updated soon or later.